components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/impl/ClientInstanceTokenBinder.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com).
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth2.token.bindings.impl;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.oauth.config.OAuthServerConfiguration;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2AccessTokenReqDTO;
+import org.wso2.carbon.identity.oauth2.model.RequestParameter;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static org.wso2.carbon.identity.oauth2.OAuth2Constants.TokenBinderType.CLIENT_INSTANCE;
+
+
+/**
+ * Client Instance Reference binding to the token.
+ */
+public class ClientInstanceTokenBinder extends AbstractTokenBinder {
+
+    private static final String CLIENT_INSTANCE_REF = "clientInstanceRef";
+
+    @Override
+    public Optional<String> getTokenBindingValue(OAuth2AccessTokenReqDTO oAuth2AccessTokenReqDTO) {
+
+        RequestParameter[] parameters = oAuth2AccessTokenReqDTO.getRequestParameters();
+        for (RequestParameter parameter : parameters) {
+            if (CLIENT_INSTANCE_REF.equals(parameter.getKey())
```

Comment:
can we verify how easy is to send this parameter in the token request using the SDk ? does SDKs allow to add extra parameter in the token request ?

Reply:
AFAIK, our SDKs do not support sending additional parameters. But wondering back-channel grant types like password/token-exchange require an SDK?